### PR TITLE
Add Koin (DI) example 

### LINKED ISSUE #10: InsertKoinIO/koin
## About the library

- *Description*:    Koin - a pragmatic lightweight dependency injection framework for Kotlin 
- *GitHub*: https://github.com/InsertKoinIO/koin
- *Official website*: https://insert-koin.io/

## What the sample could do

Definitely **not** the meaningless example of the fancy coffee machine with a thermosiphon.

See the [Kodein-DI sample](https://github.com/LouisCAD/kotlin-libraries-playground/blob/main/kotlin-jvm/src/main/kotlin/playground/KodeinDI.kt)

## Wants to contribute the sample?

- Say so in a comment here
- Follow instructions in [CONTRIBUTING.md](https://github.com/LouisCAD/kotlin-libraries-playground/blob/main/CONTRIBUTING.md)
- Submit your pull request!

### PR DESCRIPTION
**Pull request by douglascamata**
Fixes #10 and #29.

### CODE REVIEW

File: kotlin-jvm/versions.properties [hunk 1]
```diff
@@ -74,3 +74,14 @@ version.retrofit2=2.9.0
 version.sqldelight=1.4.3
 
 version.com.github.ajalt..mordant=1.2.1
+
+version.org.koin..koin-core=2.1.6
```
Comment (jmfayard) [inline kotlin-jvm/versions.properties:78]: Can you please run `$ ./gradlew refreshVersions`?
That would re-order the keys in `versions.properties` automatically.
Then remove the comment about libraries updates if it doesn't make sense to upgrade

File: kotlin-jvm/src/main/kotlin/playground/di/Koin.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    // IN the real app, we want the users we have previously stored.
+    realApp.getUser(1) shouldBe user
+}
+
+
+/**
+ * Abstraction of an application that can get and add users given a  UserRepository.
+ */
+internal class Application(environment: String) : KoinComponent {
+    private val userRepository by inject<UserRepository>(qualifier = named(environment))
```
Comment (jmfayard) [inline kotlin-jvm/src/main/kotlin/playground/di/Koin.kt:53]: I don't know about Koin, but in general I prefer to use constructor injection when that's possible. 
Is that an option here?

**Review by jmfayard — APPROVED** [review]

File: kotlin-jvm/src/main/kotlin/playground/di/Koin.kt (hunk 2 above)
Comment (douglascamata) [reply kotlin-jvm/src/main/kotlin/playground/di/Koin.kt:53]: I tried figuring out a way of using the constructor injector, but with Koin it's not as easy to achieve it as with KodeinDI :/

It requires at least 2 extra classes to achieve this. What do you think?

File: kotlin-jvm/versions.properties (hunk 1 above)
Comment (douglascamata) [reply kotlin-jvm/versions.properties:78]: Ahhh, nice. I'll run it. The comment with the version appeared here automagically o_o

File: kotlin-jvm/src/main/kotlin/playground/di/Koin.kt (hunk 2 above)
Comment (jmfayard) [reply kotlin-jvm/src/main/kotlin/playground/di/Koin.kt:53]: Ok, leave it as is for now!

File: kotlin-jvm/versions.properties (hunk 1 above)
Comment (jmfayard) [reply kotlin-jvm/versions.properties:78]: @douglascamata 
Yes, that's the main feature of https://github.com/jmfayard/refreshVersions
It lookup available dependencies updates for you, put it as comment in a way that it makes it easy to upgrade.
But the comment are not meant to be kept in git

You will have to merge `origin/main`, there is a (hopefully simple to resolve) merge conflict

**Review by jmfayard — APPROVED** [review]

File: kotlin-jvm/versions.properties (hunk 1 above)
Comment (jmfayard) [reply kotlin-jvm/versions.properties]: Resolved, I took care of the merge conflicts

**Comment by jmfayard** [comment]
I merged it, thanks Douglas!

**Pull request merged**